src/main/java/com/cegeka/academy/service/invitation/InvitationServiceImpl.java

```diff
@@ -151,4 +155,39 @@ public void sendGroupInvitationsToPrivateEvents(Long idGroup, Invitation invitat
         }
     }
 
+    @Override
+    public Invitation createChallengeInvitationForOneUser(InvitationDTO invitationDTO, Long challengeId)
+            throws ExistingItemException, NotFoundException {
+
+        long userId = invitationDTO.getUserId();
+
+        if (userChallengeRepository.findByUserIdAndChallengeId(userId, challengeId).isPresent()) {
+            throw new ExistingItemException();
+        } else {
+            Invitation invitation = invitationRepository.save(InvitationMapper.convertInvitationDTOToInvitation(invitationDTO));
+            UserChallenge userChallenge = new UserChallenge();
+
+            Challenge challenge = challengeRepository.findById(challengeId).orElseThrow(() -> new NotFoundException().setMessage("Challenge not found"));
+
+            userChallenge.setChallenge(challenge);
+            userChallenge.setPoints(0);
+            userChallenge.setUser(invitation.getUser());
+            userChallenge.setInvitation(invitation);
+            userChallenge.setStatus("Not started");
+            userChallenge.setChallengeAnswer(null);
+            userChallenge.setStartTime(new Date());
+            userChallenge.setEndTime(new Date());
+
+            userChallengeService.createUserChallenge(userChallenge);
+
```

Comment:
Toata logica de creare a unui UserChallenge ar trebui sa apartina metodei createUserChallenge. Eventual poti denumi aceasta metoda  "initiateUserChallenge" astfel incat sa-i dai un nume potrivit pentru ceea ce doresti sa faci, si anume: initializarea unui UserChallenge. Exista posibilitatea ca pe viitor sa avem o alta modalitate de construire a unui UserChallenge (spre ex cu informatii din UI), iar in cazul acesta o sa fie nevoie de o alta metoda.